pombola/south_africa/views/attendance.py

```diff
@@ -49,36 +49,62 @@ def filter_attendance(self, annual_attendance, party, position):
         Filter meeting attendance to only include items which match
         the party and position selections by the user.
         """
+
         attendance_summary = annual_attendance['attendance_summary']
         if party:
             attendance_summary = [ma for ma in attendance_summary if
                 ma['member']['party_name'] == party]
 
         year_end_date = datetime.datetime.strptime(
             annual_attendance['end_date'], "%Y-%m-%d").date()
-        active_ministers = Position.objects.filter(
+        today = datetime.date.today()
+
+        active_minister_positions = Position.objects.filter(
                 Q(title__slug__startswith='minister') |
                 Q(title__slug__startswith='deputy-minister')
             ).currently_active(
-                year_end_date)
+                today if year_end_date > today else year_end_date).select_related('person')
 
-        active_minister_slugs = set(am.person.slug for am in active_ministers)
+        active_minister_slugs = set(am.person.slug for am in active_minister_positions)
 
-        minister_attendance = [ma for ma in attendance_summary if
-                ma['member']['pa_url'] and
-                ma['member']['pa_url'].split('/')[-2] in active_minister_slugs]
-
-        minister_ids = set(m['member']['id'] for m in minister_attendance)
+        minister_attendance = []
+        for attendance in attendance_summary:
+            if attendance['member']['pa_url']:
+                if attendance['member']['pa_url'].split('/')[-2] in active_minister_slugs:
+                    minister_attendance.append(attendance)
+                    attendance_summary.remove(attendance)
+                    active_minister_slugs.remove(attendance['member']['pa_url'].split('/')[-2])
 
         if position == 'ministers':
-            attendance_summary = minister_attendance
+            # Some Ministers don't have attendance records,
+            # and we need to show that they haven't attendended any meetings
+            pa_url = 'http://www.pa.org.za'
```

Comment:
Ah, yes. That's much better doing it this way around. It was for consistency that I prepended it.